libtransmission/port-forwarding-upnp.cc

```diff
@@ -295,23 +306,27 @@ tr_port_forwarding_state tr_upnpPulse(tr_upnp* handle, tr_port port, bool is_ena
         freeUPNPDevlist(devlist);
     }
 
-    if ((handle->state == UpnpState::Idle) && (handle->isMapped) && (!is_enabled || handle->port != port))
+    if (handle->state <= UpnpState::Failed && handle->isMapped &&
```

Comment:
For maintainability, enumerations shouldn't have ordered comparisons (since new enumeration values could be inserted or re-ordered, causing side-effects). Also that makes it harder to read/review for humans, since we don't know the affected enumeration values without looking them up.

I propose:
```suggestion
    if ((handle->state == UpnpState::Idle || handle->state == UpnpState::Failed) && handle->isMapped &&
```

Reply:
6612dd5